### DISCUSSION TITLE
Async issues

### DISCUSSION THREAD

**Discussion by jasonhutton-sd** [JohannesKlauss/react-hotkeys-hook#693] [Q&A]
Calling a function that is using async and await and it does not appear to be working. Any ideas?

**Comment by JohannesKlauss** [comment]
Please provide a bit more context, preferably a codesandbox so I can reproduce the issue. Are you calling a async function inside the callback or is you callback itself async? Also: What exactly isn't working?

**Reply by jasonhutton-sd** [reply]
I can't really share too much but let's try.

I had a function that was like this:
`const deleteItem = async () => {
    const result = await serverCall();
    ... do something with the results
   }
`
Then when I would use the use the following
`useHotkeys('ctrl+d', () => deleteItem());`

It would not have the results available and would fail the follow up actions as the results were null.
I also tried using
`useHotkeys('ctrl+d', async () => await deleteItem());`

Same result. When I do a direct onKeyDown event and handler it works great. I think you have to somehow account for the async/await to bubble up though the react-hotkeys code.

Thanks
Jason

**Reply by JohannesKlauss** [reply]
I can't really see how that would relate to the hook, since it's just syntactic sugar around promise fulfillment. 
Can you build up a codesandbox that illustrates the issue?

**Comment by maryghermann** [comment]
I'm experiencing this same issue

**Reply by JohannesKlauss** [reply]
As said above:
I can't really see how that would relate to the hook, since it's just syntactic sugar around promise fulfillment.
Can you build up a codesandbox that illustrates the issue?
